mempool/reactor.go

```diff
@@ -168,6 +164,27 @@ func (memR *Reactor) Receive(e p2p.Envelope) {
 	// broadcasting happens from go routines per peer
 }
 
+// TryAddTx attempts to add an incoming transaction to the mempool.
+// When the sender is nil, it means the transaction comes from an RPC endpoint.
+func (memR *Reactor) TryAddTx(tx types.Tx, sender p2p.Peer) (*abcicli.ReqRes, error) {
+	senderID := noSender
+	if sender != nil {
+		senderID = sender.ID()
+	}
+
+	reqRes, err := memR.mempool.CheckTx(tx, senderID)
+	switch {
+	case errors.Is(err, ErrTxInCache):
+		memR.Logger.Debug("Tx already exists in cache", "tx", log.NewLazySprintf("%v", tx.Hash()))
```

Comment:
There's a change in behaviour with this code. These messages were logged only in `Receive`, for transactions coming from peers. Now we log them also when transactions coming via RPC endpoints.

Reply:
You might want to add the sender to the log